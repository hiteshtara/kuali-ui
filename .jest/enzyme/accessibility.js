/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
