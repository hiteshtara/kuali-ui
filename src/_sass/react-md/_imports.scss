@import '~react-md/src/scss/react-md';

$md-font-regular: null;

@include react-md-typography;
@include react-md-accessibility;
@include react-md-grid;
@include react-md-transitions;
@include react-md-collapsers;
@include react-md-inks;
@include react-md-overlays;
@include react-md-subheaders;
@include react-md-colors;
