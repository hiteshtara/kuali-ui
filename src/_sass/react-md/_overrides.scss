.md-text--theme-primary,
.md-picker-text--active {
  color: #000;
}
