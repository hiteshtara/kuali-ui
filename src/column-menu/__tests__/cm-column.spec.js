/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

/* eslint-disable no-undef */
import React, { Component } from 'react'
import { mount, shallow } from 'enzyme'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'
import { cloneDeep } from 'lodash'

import ColumnMenuColumn from '../cm-column'
import { ListItem } from 'react-md'

describe.skip('Column Menu Column', () => {
  let sandbox
  let props

  beforeEach(() => {
    sandbox = sinon.sandbox.create()
    props = {
      addColumn: jest.fn(),
      data: ['one', 'two', 'three'],
      filter: true,
      match: ['one', 'two', 'three'],
      onSelect: jest.fn()
    }
  })

  afterEach(() => {
    sandbox.restore()
  })

  test('snapshot matches', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  test('returns default styling', () => {
    props.style = {}
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()

    expect(inst.getStyles()).toEqual({
      menu: {
        overflowY: 'auto',
        width: 180,
        height: 352
      }
    })
  })

  test('getItems not called if props are unchanged', () => {
    const getItemsSpy = sandbox.spy(ColumnMenuColumn.prototype, 'getItems')
    const wrapper = mount(<ColumnMenuColumn {...props} />)

    expect(getItemsSpy.callCount).toBe(1)
    wrapper.setProps(props)
    expect(getItemsSpy.callCount).toBe(1)
  })

  test('inits data only when necessary', () => {
    const newProps = cloneDeep(props)
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()

    expect(inst.shouldInitData(props, props)).toBe(false)

    props.data = ['array']
    newProps.data = { match: true }
    expect(inst.shouldInitData(props, newProps)).toBe(true)

    props.data = ['one', 'two', 'three']
    newProps.data = ['one', 'two', 'three']
    expect(inst.shouldInitData(props, newProps)).toBe(false)
    newProps.data = ['one', 'two', '-NOT THREE!-']
    expect(inst.shouldInitData(props, newProps)).toBe(true)

    props.data = { match: true }
    newProps.data = { match: true }
    expect(inst.shouldInitData(props, newProps)).toBe(false)
    newProps.data = { match: false }
    expect(inst.shouldInitData(props, newProps)).toBe(true)

    props.data = () => 'match'
    newProps.data = () => 'match'
    expect(inst.shouldInitData(props, newProps)).toBe(true)
    newProps.data = () => '-NOT A MATCH!-'
    expect(inst.shouldInitData(props, newProps)).toBe(true)
  })

  test('get items: data type array', done => {
    props.filter = false
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual(props.data)
      done()
    })
  })

  test('get filtered items: data type array', done => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()

    inst.setState({ filter: 'thre' })
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual(['three'])
      done()
    })
  })

  test('get items: data type object', done => {
    props.filter = false
    props.data = { one: 1, two: 2, three: 3 }
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual([1, 2, 3])
      done()
    })
  })

  test('get items: data type object with meta data', done => {
    props.filter = false
    props.data = {
      _id: 'ayU8pkSs',
      _name: 'name',
      _type: 'type',
      one: 1,
      two: 2,
      three: 3
    }
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual([1, 2, 3])
      expect(inst.state.meta).toEqual({
        id: 'ayU8pkSs',
        name: 'name',
        type: 'type'
      })
      done()
    })
  })

  test('get filtered items: data type object', done => {
    props.data = { one: 1, two: 2, three: 3 }
    props.filter = true
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()

    inst.setState({ filter: 'thre' })
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual([3])
      done()
    })
  })

  test('get items: data type function', done => {
    props.filter = false
    props.data = (filter, cb) => {
      cb(null, [1, 2, 3])
    }
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual([1, 2, 3])
      done()
    })
  })

  test('get items: data type function with meta data', done => {
    props.filter = false
    props.data = (filter, cb) => {
      cb({ id: 'ayU8pkSs', name: 'name', type: 'type' }, [1, 2, 3])
    }
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual([1, 2, 3])
      expect(inst.state.meta).toEqual({
        id: 'ayU8pkSs',
        name: 'name',
        type: 'type'
      })
      done()
    })
  })

  test('get filtered items: data type function', done => {
    props.data = (filter, cb) => {
      cb({ id: 'ayU8pkSs', name: 'name', type: 'type' }, [3])
    }
    props.filter = true
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()

    inst.setState({ filter: 'thre' })
    inst.getItems().then(() => {
      expect(inst.state.items).toEqual([3])
      done()
    })
  })

  test('get items: data type string', done => {
    props.data = 'string'
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()

    inst.getItems().catch(err => {
      expect(err.message).toMatch('Unexpected data type')
      done()
    })
  })

  test('can get an item by index', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    expect(wrapper.instance().getItem(1)).toEqual('two')
  })

  test('provides an isLeaf function that always returns undefined', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    expect(wrapper.instance().isLeaf()).toBe(undefined)
  })

  test('can call isType but will always be false', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    expect(wrapper.instance().isType('cabbage')).toBe(false)
  })

  test('can call containsType but will always be true', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    expect(wrapper.instance().containsType('cabbage')).toBe(true)
  })

  test('can add a default Column', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    wrapper.instance().addColumn('key', 'value')
    expect(props.addColumn.mock.calls.length).toBe(1)
    expect(props.addColumn.mock.calls[0][0].displayName).toEqual(
      'ColumnMenuColumn'
    )
    expect(props.addColumn.mock.calls[0][1]).toEqual('key')
    expect(props.addColumn.mock.calls[0][2]).toEqual('value')
  })

  test('can add a custom Column', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)

    class Custom extends Component {
      static displayName = 'Custom';
      render () {
        return <div>Custom</div>
      }
    }

    wrapper.instance().addColumn('key', { config: Custom })
    expect(props.addColumn.mock.calls.length).toBe(1)
    expect(props.addColumn.mock.calls[0][0].displayName).toEqual('Custom')
    expect(props.addColumn.mock.calls[0][1]).toEqual('key')
    expect(props.addColumn.mock.calls[0][2]).toEqual({ config: Custom })
  })

  test('can update the filter', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.filterChangeHandler('cabbage')
    expect(inst.state.filter).toEqual('cabbage')
  })

  test('can select an item in the column', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.onSelect(null, 1)
    expect(inst.state.selected).toEqual('two')
  })

  test('can select an item in the column that is a leaf node', () => {
    props.data = [{ _cmc: { leaf: true }, _name: 'Phillip' }]
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    inst.onSelect(null, 0)
    expect(inst.state.selected).toEqual(props.data[0])
  })

  test('can click an item', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const $items = wrapper.find('.column-menu-item div[role="button"]')
    expect($items.length).toBe(3)
    $items.at(1).simulate('click')
    expect(wrapper.instance().state.selected).toEqual('two')
  })

  test('noop does nothing', () => {
    const wrapper = shallow(<ColumnMenuColumn {...props} />)
    wrapper.instance().noop()
  })

  test('can provide a custom ListItem', () => {
    props.listItem = class Custom extends ListItem {
      render () {
        return <div className='custom-list-item' />
      }
    }
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    expect(wrapper.find('.custom-list-item').length).toBe(3)
  })

  test('can render menu items', () => {
    const wrapper = mount(<ColumnMenuColumn {...props} />)
    const inst = wrapper.instance()
    const component = inst.renderColumnMenuItem('v', 'k', true, ListItem)
    expect(component.props.primaryText).toEqual('k')
  })
})
