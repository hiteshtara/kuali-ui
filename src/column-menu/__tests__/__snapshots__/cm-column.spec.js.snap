// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Menu Column snapshot matches 1`] = `
<ColumnMenuColumn
  addColumn={[Function]}
  cache={true}
  data={
    Array [
      "one",
      "two",
      "three",
    ]
  }
  filter={true}
  match={
    Array [
      "one",
      "two",
      "three",
    ]
  }
  onSelect={[Function]}
>
  <div
    className="col"
  >
    <ColumnMenuColumnHeader
      filter={true}
      filterText=""
      onChange={[Function]}
    >
      <div
        className="hdr"
      >
        <TextField
          fullWidth={false}
          id="cmc-filter-field"
          leftIconStateful={true}
          lineDirection="left"
          onChange={[Function]}
          passwordIconChildren="remove_red_eye"
          placeholder="Filter"
          rightIcon={
            <FontIcon
              iconClassName="material-icons"
              style={
                Object {
                  "left": "-50px",
                  "position": "absolute",
                  "top": "10px",
                }
              }
            >
              action_search
            </FontIcon>
          }
          rightIconStateful={true}
          style={
            Object {
              "left": "13px",
              "position": "absolute",
              "top": "-3px",
              "width": "160px",
            }
          }
          type="text"
          value=""
        >
          <div
            className="md-text-field-container md-inline-block md-text-field-container--input"
            onClick={[Function]}
            style={
              Object {
                "height": null,
                "left": "13px",
                "position": "absolute",
                "top": "-3px",
                "width": "160px",
              }
            }
          >
            <FloatingLabel
              active={false}
              error={false}
              floating={false}
              htmlFor="cmc-filter-field"
              iconOffset={false}
            />
            <div
              className="md-text-field-icon-container"
            >
              <div
                className="md-text-field-divider-container"
              >
                <InputField
                  fullWidth={false}
                  id="cmc-filter-field"
                  inlineIndicator={false}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onHeightChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Filter"
                  type="text"
                  value=""
                >
                  <input
                    className="md-text-field md-text md-text-field--margin"
                    id="cmc-filter-field"
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Filter"
                    type="text"
                    value=""
                  />
                </InputField>
                <TextFieldDivider
                  active={false}
                  error={false}
                  lineDirection="left"
                >
                  <Divider
                    className="md-divider--text-field md-divider--expand-from-left"
                  >
                    <hr
                      className="md-divider md-divider--text-field md-divider--expand-from-left"
                    />
                  </Divider>
                </TextFieldDivider>
              </div>
              <FontIcon
                className="md-text-field-icon"
                iconClassName="material-icons"
                style={
                  Object {
                    "left": "-50px",
                    "position": "absolute",
                    "top": "10px",
                  }
                }
              >
                <i
                  className="md-icon material-icons md-text-field-icon"
                  style={
                    Object {
                      "left": "-50px",
                      "position": "absolute",
                      "top": "10px",
                    }
                  }
                >
                  action_search
                </i>
              </FontIcon>
            </div>
            <TextFieldMessage
              active={false}
              currentLength={0}
              error={false}
              leftIcon={false}
              rightIcon={true}
            />
          </div>
        </TextField>
      </div>
    </ColumnMenuColumnHeader>
    <Menu
      className="column-menu-menu"
      component="div"
      fullWidth={true}
      isOpen={true}
      listClassName="column-menu-list"
      onClose={[Function]}
      position="tr"
      transitionEnterTimeout={200}
      transitionLeaveTimeout={200}
      transitionName="md-menu"
    >
      <ReactCSSTransitionGroup
        aria-expanded={true}
        aria-haspopup={true}
        className="md-inline-block md-menu-container md-full-width column-menu-menu"
        component="div"
        transitionAppear={false}
        transitionEnter={true}
        transitionEnterTimeout={200}
        transitionLeave={true}
        transitionLeaveTimeout={200}
        transitionName="md-menu"
      >
        <ReactTransitionGroup
          aria-expanded={true}
          aria-haspopup={true}
          childFactory={[Function]}
          className="md-inline-block md-menu-container md-full-width column-menu-menu"
          component="div"
          transitionAppear={false}
          transitionEnter={true}
          transitionEnterTimeout={200}
          transitionLeave={true}
          transitionLeaveTimeout={200}
          transitionName="md-menu"
        >
          <div
            aria-expanded={true}
            aria-haspopup={true}
            className="md-inline-block md-menu-container md-full-width column-menu-menu"
          >
            <ReactCSSTransitionGroupChild
              appear={false}
              enter={true}
              enterTimeout={200}
              leave={true}
              leaveTimeout={200}
              name="md-menu"
            >
              <List
                className="column-menu-list"
                onClick={[Function]}
              >
                <ul
                  className="md-list md-list--menu md-list--menu-scrollable md-list--menu-tr column-menu-list"
                  onClick={[Function]}
                >
                  <ListItem
                    active={false}
                    activeClassName="md-text--theme-primary"
                    className="column-menu-item"
                    component="div"
                    data-test="cm-item-one"
                    expanderIconChildren="keyboard_arrow_down"
                    onClick={[Function]}
                    primaryText="one"
                    value={0}
                  >
                    <li
                      className="md-list-item column-menu-item"
                    >
                      <AccessibleFakeButtonInked
                        __SUPER_SECRET_REF__={[Function]}
                        aria-expanded={null}
                        className="md-list-tile md-text"
                        component="div"
                        data-test="cm-item-one"
                        inkTransitionEnterTimeout={450}
                        inkTransitionLeaveTimeout={300}
                        inkTransitionOverlap={150}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        value={0}
                      >
                        <AccessibleFakeButton
                          aria-expanded={null}
                          className="md-list-tile md-text"
                          component="div"
                          data-test="cm-item-one"
                          ink={
                            <InkContainer
                              className={undefined}
                              disabledInteractions={undefined}
                              inkClassName={undefined}
                              inkStyle={undefined}
                              style={undefined}
                              transitionEnterTimeout={450}
                              transitionLeaveTimeout={300}
                              transitionOverlap={150}
                              waitForInkTransition={undefined}
                            />
                          }
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                          value={0}
                        >
                          <div
                            aria-expanded={null}
                            aria-pressed={false}
                            className="md-fake-btn md-pointer--hover md-list-tile md-text"
                            data-test="cm-item-one"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseLeave={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}
                            value={0}
                          >
                            <InkContainer
                              transitionEnterTimeout={450}
                              transitionLeaveTimeout={300}
                              transitionOverlap={150}
                            >
                              <ReactTransitionGroup
                                childFactory={[Function]}
                                className="md-ink-container"
                                component="div"
                              >
                                <div
                                  className="md-ink-container"
                                />
                              </ReactTransitionGroup>
                            </InkContainer>
                            <TileAddon
                              active={false}
                              activeClassName="md-text--theme-primary"
                            />
                            <ListItemText
                              active={false}
                              activeClassName="md-text--theme-primary"
                              className=""
                              primaryText="one"
                            >
                              <div
                                className="md-tile-content"
                              >
                                <div
                                  className="md-tile-text--primary md-text"
                                >
                                  one
                                </div>
                              </div>
                            </ListItemText>
                            <TileAddon
                              active={false}
                              activeClassName="md-text--theme-primary"
                            />
                          </div>
                        </AccessibleFakeButton>
                      </AccessibleFakeButtonInked>
                    </li>
                  </ListItem>
                  <ListItem
                    active={false}
                    activeClassName="md-text--theme-primary"
                    className="column-menu-item"
                    component="div"
                    data-test="cm-item-two"
                    expanderIconChildren="keyboard_arrow_down"
                    onClick={[Function]}
                    primaryText="two"
                    value={1}
                  >
                    <li
                      className="md-list-item column-menu-item"
                    >
                      <AccessibleFakeButtonInked
                        __SUPER_SECRET_REF__={[Function]}
                        aria-expanded={null}
                        className="md-list-tile md-text"
                        component="div"
                        data-test="cm-item-two"
                        inkTransitionEnterTimeout={450}
                        inkTransitionLeaveTimeout={300}
                        inkTransitionOverlap={150}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        value={1}
                      >
                        <AccessibleFakeButton
                          aria-expanded={null}
                          className="md-list-tile md-text"
                          component="div"
                          data-test="cm-item-two"
                          ink={
                            <InkContainer
                              className={undefined}
                              disabledInteractions={undefined}
                              inkClassName={undefined}
                              inkStyle={undefined}
                              style={undefined}
                              transitionEnterTimeout={450}
                              transitionLeaveTimeout={300}
                              transitionOverlap={150}
                              waitForInkTransition={undefined}
                            />
                          }
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                          value={1}
                        >
                          <div
                            aria-expanded={null}
                            aria-pressed={false}
                            className="md-fake-btn md-pointer--hover md-list-tile md-text"
                            data-test="cm-item-two"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseLeave={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}
                            value={1}
                          >
                            <InkContainer
                              transitionEnterTimeout={450}
                              transitionLeaveTimeout={300}
                              transitionOverlap={150}
                            >
                              <ReactTransitionGroup
                                childFactory={[Function]}
                                className="md-ink-container"
                                component="div"
                              >
                                <div
                                  className="md-ink-container"
                                />
                              </ReactTransitionGroup>
                            </InkContainer>
                            <TileAddon
                              active={false}
                              activeClassName="md-text--theme-primary"
                            />
                            <ListItemText
                              active={false}
                              activeClassName="md-text--theme-primary"
                              className=""
                              primaryText="two"
                            >
                              <div
                                className="md-tile-content"
                              >
                                <div
                                  className="md-tile-text--primary md-text"
                                >
                                  two
                                </div>
                              </div>
                            </ListItemText>
                            <TileAddon
                              active={false}
                              activeClassName="md-text--theme-primary"
                            />
                          </div>
                        </AccessibleFakeButton>
                      </AccessibleFakeButtonInked>
                    </li>
                  </ListItem>
                  <ListItem
                    active={false}
                    activeClassName="md-text--theme-primary"
                    className="column-menu-item"
                    component="div"
                    data-test="cm-item-three"
                    expanderIconChildren="keyboard_arrow_down"
                    onClick={[Function]}
                    primaryText="three"
                    value={2}
                  >
                    <li
                      className="md-list-item column-menu-item"
                    >
                      <AccessibleFakeButtonInked
                        __SUPER_SECRET_REF__={[Function]}
                        aria-expanded={null}
                        className="md-list-tile md-text"
                        component="div"
                        data-test="cm-item-three"
                        inkTransitionEnterTimeout={450}
                        inkTransitionLeaveTimeout={300}
                        inkTransitionOverlap={150}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        value={2}
                      >
                        <AccessibleFakeButton
                          aria-expanded={null}
                          className="md-list-tile md-text"
                          component="div"
                          data-test="cm-item-three"
                          ink={
                            <InkContainer
                              className={undefined}
                              disabledInteractions={undefined}
                              inkClassName={undefined}
                              inkStyle={undefined}
                              style={undefined}
                              transitionEnterTimeout={450}
                              transitionLeaveTimeout={300}
                              transitionOverlap={150}
                              waitForInkTransition={undefined}
                            />
                          }
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                          value={2}
                        >
                          <div
                            aria-expanded={null}
                            aria-pressed={false}
                            className="md-fake-btn md-pointer--hover md-list-tile md-text"
                            data-test="cm-item-three"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseLeave={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}
                            value={2}
                          >
                            <InkContainer
                              transitionEnterTimeout={450}
                              transitionLeaveTimeout={300}
                              transitionOverlap={150}
                            >
                              <ReactTransitionGroup
                                childFactory={[Function]}
                                className="md-ink-container"
                                component="div"
                              >
                                <div
                                  className="md-ink-container"
                                />
                              </ReactTransitionGroup>
                            </InkContainer>
                            <TileAddon
                              active={false}
                              activeClassName="md-text--theme-primary"
                            />
                            <ListItemText
                              active={false}
                              activeClassName="md-text--theme-primary"
                              className=""
                              primaryText="three"
                            >
                              <div
                                className="md-tile-content"
                              >
                                <div
                                  className="md-tile-text--primary md-text"
                                >
                                  three
                                </div>
                              </div>
                            </ListItemText>
                            <TileAddon
                              active={false}
                              activeClassName="md-text--theme-primary"
                            />
                          </div>
                        </AccessibleFakeButton>
                      </AccessibleFakeButtonInked>
                    </li>
                  </ListItem>
                </ul>
              </List>
            </ReactCSSTransitionGroupChild>
          </div>
        </ReactTransitionGroup>
      </ReactCSSTransitionGroup>
    </Menu>
  </div>
</ColumnMenuColumn>
`;
